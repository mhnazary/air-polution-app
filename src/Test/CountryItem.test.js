import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import CountriesItem from '../components/CountryItem';

// Create a mock Redux store
test('renders component with correct data', () => {
  const component = renderer.create(
    <MemoryRouter>
      <CountriesItem
        data={{ latitude: 20, longitude: 10 }}
        imageUrl="image-url"
        country="Country"
        capital="Capital"
      />
    </MemoryRouter>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
