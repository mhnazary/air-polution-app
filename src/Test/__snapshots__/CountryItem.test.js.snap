// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component with correct data 1`] = `
<li
  className="country-list"
>
  <div
    className="flex justify-between px-8 items-center"
  >
    <span
      className="text-base font-normal"
    >
      Lat: 46.9479
    </span>
    <span
      className="text-base font-normal"
    >
      Lon: 7.4474
    </span>
    <a
      className="country-data"
      href="/stats?lat=46.9479&lon=7.4474"
      onClick={[Function]}
    >
      <svg
        className=" text-2xl"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </a>
  </div>
  <div>
    <img
      alt="Switzerland"
      className="country-shape"
      src="https://raw.githubusercontent.com/Ginohmk/worldMaps/main/maps/ch/vector.svg"
    />
  </div>
  <div>
    <h3
      className="text-2xl font-bold pl-8"
    >
      Switzerland
    </h3>
    <p
      className="text-lg font-bold pl-8"
    >
      Bern
    </p>
  </div>
</li>
`;
